runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -1051,7 +1061,8 @@ else if (!isCorsRequestAllowed(server.binding, headers))
                         }
 
                         HttpRouteConfig route = binding.resolve(exchangeAuth, headers::get);
-                        if (route != null)
+                        server.selectRequest(headers::get);
+                        if (route != null && server.validate(beginEx, headers))
```

Comment:
The `validate` method can also move to `HttpBindingConfig`, then the `selectRequest` / `resolveRequest` and supporting parse methods would be internal with only on-stack state (no fields) needed.